cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -486,13 +545,16 @@ private File getExecutorPodTemplateFile() throws Exception {
       container.addVolumeMountsItem(new V1VolumeMount().name(configMapName)
                                       .mountPath(CDAP_LOCALIZE_FILES_PATH).readOnly(true));
     }
+    if (workloadIdentityEnabled) {
+      setupWorkloadIdentityForPodSpec(executorPodSpec, workloadIdentityPool);
+    }
 
     // Create spark template file. We do not delete it because pod will get deleted at the end of job completion.
     return serializePodTemplate(executorPod);
   }
 
   private File serializePodTemplate(V1Pod v1Pod) throws IOException {
-    File templateFile = new File(POD_TEMPLATE_FILE_NAME + UUID.randomUUID().toString());
```

Comment:
It's just creating a file in the current directory? Is it expected? I mean, you never knows if current dir is writable

Reply:
As far as I can tell, that's correct. However, the working directory is set by the image which the pod is launched with though, so unless it is explicitly overwritten it should deterministic.

Reply:
Yep. Weird, but let's not change it.